stock_valuation_fifo_lot/models/stock_move.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2023 Ecosoft Co., Ltd (https://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _prepare_common_svl_vals(self):
+        """
+        Prepare lots/serial numbers on stock valuation report
+        """
+        self.ensure_one()
+        res = super()._prepare_common_svl_vals()
+        res.update(
+            {
+                "lot_ids": [(6, 0, self.lot_ids.ids)],
+            }
+        )
+        return res
+
+    def _create_out_svl(self, forced_quantity=None):
+        """
+        Send context current move to _create_out_svl function
+        """
+        layers = self.env["stock.valuation.layer"]
+        for move in self:
+            move = move.with_context(used_in_move_id=move.id)
+            layer = super(StockMove, move)._create_out_svl(
+                forced_quantity=forced_quantity
+            )
+            layers |= layer
+        return layers
+
+    def _create_in_svl(self, forced_quantity=None):
+        """
+        1. Check stock move - Multiple lot on the stock move is not
+           allowed for incoming transfer
+        2. Change product standard price to first available lot price
+        """
+        layers = self.env["stock.valuation.layer"]
+        for move in self:
+            layer = super(StockMove, move)._create_in_svl(
+                forced_quantity=forced_quantity
+            )
+            # Calculate standard price (Sorted by lot created date)
+            if (
+                move.product_id.cost_method == "fifo"
+                and move.product_id.tracking != "none"
+            ):
+                all_candidates = move.product_id._get_all_candidates(
+                    move.company_id, sort_by="lot_create_date"
+                )
+                if all_candidates:
+                    move.product_id.sudo().with_company(
+                        move.company_id.id
+                    ).with_context(
+                        disable_auto_svl=True
+                    ).standard_price = all_candidates[
+                        0
+                    ].unit_cost
+            layers |= layer
+        return layers
```

Comment:
@AungKoKoLin1997 I actually have no idea what this adjustment is for. Can you please explain what it is.

Reply:
This adjustment is for using first lot product price when we receive this product.
Eg,
- Receive product with lot number (11111) for price 100
- Receive product with lot number (22222) for price 200
Now, the standard price of product is 100.
Then,
- Deliver product with lot number(11111).
After that, the standard price of product is 200.
We create another receipt.
- Receive product with lot number (11111) for price 300
We want to assign this price as standard cost because this lot is created first. So, the standard price is 300 with current design. The odoo standard behavior will keep 200 as standard price.

Reply:
> - Receive product with lot number (11111) for price 300

Is the assumption here to change the price of the PO line for the first receipt of this lot from 100 (or 200) to 300, and then receive on the same stock move line as the first receipt by increasing the done quantity (sounds like a corner-case scenario to me)?

> We want to assign this price as standard cost because this lot is created first. So, the standard price is 300 with current design. The odoo standard behavior will keep 200 as standard price.

I fail to see the point of doing this manipulation. What business goal does it try to address?